src/cargo/commands.rs

```diff
@@ -0,0 +1,168 @@
+
+    const ERR_MSG : &str = "Expected manifest_path to point to Cargo.toml";
+
+    if let Some(file_name) = manifest_path.file_name() {
+
+        if file_name != "Cargo.toml" {
+            bail!(ERR_MSG);
+        }
+
+    } else {
+        bail!(ERR_MSG);
+    }
+
+    Ok(())
+}
+
+
+#[cfg(test)]
+mod tests {
+    pub use super::*;
+    pub use std::ffi::OsString;
+
+    mod retrieve_metadata {
+        pub use super::*;
+
+        #[test]
+        fn it_exists() {
+            // arrange
+            let path = Path::new(".");
+            // act
+            retrieve_metadata(path);
+            // assert
+        }
+
+        #[test]
+        fn it_should_add_the_manifest_path_in_arguments() {
+            // arrange
+            let path = Path::new("my-manifest-path");
+            // act
+            let cmd = retrieve_metadata(path);
+            // assert
+            assert!(cmd.args.contains(&OsString::from("my-manifest-path")))
+        }
+    }
+
+    mod generate_analysis {
+        use super::*;
+
+        #[test]
+        fn it_exists() {
+            // arrange
+            // act
+            let res = generate_analysis(&PathBuf::from("Cargo.toml"),false);
+            // assert
+            assert!(res.is_ok())
+        }
+
+        #[test]
+        fn it_should_add_verbose_flag_when_verbosity_is_enabled() {
+            // arrange
+            // act
+            let res = generate_analysis(&PathBuf::from("Cargo.toml"), true).unwrap();
+            // assert
+            assert!(res.args.contains(&OsString::from("--verbose")))
+        }
+
+        #[test]
+        fn it_should_err_if_manifest_path_doesnt_point_to_cargo_toml() {
+            // arrange
+            let manifest_path = PathBuf::from("Hello.txt");
+            // act
+            let res = generate_analysis(&manifest_path, false);
+            // assert
+            assert!(res.is_err())
+        }
+    }
+
+    mod check_manifest_path_points_to_cargo_toml {
+        use super::*;
+
+        #[test]
+        fn it_exists() {
+            // arrange
+            let path = PathBuf::from("Cargo.toml");
+            // act
+            let res = check_manifest_path_points_to_cargo_toml(&path);
+            // assert
+            assert!(res.is_ok())
+        }
+
+        #[test]
+        fn it_err_on_directories() {
+            // arrange
+            let path = PathBuf::from("/users/rusty/");
+            // act
+            let res = check_manifest_path_points_to_cargo_toml(&path);
+            // assert
+            assert!(res.is_err())
+        }
+
+        #[test]
+        fn it_returns_err_if_filename_is_not_cargo_toml() {
+            // arrange
+            let path = PathBuf::from("/users/rusty/Rusty.toml");
+            // act
+            let res = check_manifest_path_points_to_cargo_toml(&path);
+            // assert
+            assert!(res.is_err())
+        }
+    }
+}
```

Comment:
Is this whole file new because it got moved?

Reply:
nope, the file is new. I extracted the logic of command building here and I want to unit test it

Reply:
Even better. I'm all for more tests. Looking forward to seeing the end result :D